src/main/java/com/minecolonies/core/client/gui/WindowPostBox.java

```diff
@@ -97,6 +100,75 @@ public WindowPostBox(final AbstractBuildingView buildingView)
                 this.tick = 10;
             }
         });
+
+        int offset = 0;
+        final Random random = new Random(buildingView.getID().hashCode());
+        boolean anyVisible = false;
+
+        for (IBuildingModuleView view : buildingView.getAllModuleViews())
+        {
+            if (view.isPageVisible())
+            {
+                anyVisible = true;
+                break;
+            }
+        }
+
+
+        if (buildingView.getAllModuleViews().size() > 0 && anyVisible)
```

Comment:
If that code is exactly the same as in the other class, we could make this a utility method and call the utility method from both places

Reply:
the offsets are slightly different, but I can add that as a parameter